### PR TITLE
Upgrading scala 2.12

### PR DESCRIPTION
**Pull request by laurence-bird**

### CODE REVIEW

File: app/controllers/AuthActions.scala [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
+
+trait AuthActions { _: GoogleAuthComponents =>
 
   /**
     * Should authentication be enabled.
     * This should only be set to false when running service tests.
     * If false, authentication will always succeed, with a dummy user added to the request.
     */
-  def enableAuth: Boolean
+  val enableAuth: Boolean
```
Comment (filosganga) [inline app/controllers/AuthActions.scala]: Any reason to change it to val?

File: app/controllers/MockAuthRequest.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.slf4j.MDC
+import play.api.Logger
+import play.api.mvc.{Action, ActionTransformer, Request}
+import Auth.AuthRequest
+import com.gu.googleauth.UserIdentity
+import play.api.mvc.Security.AuthenticatedRequest
+
+import scala.concurrent.Future
+
+object MockAuthRequest extends ActionTransformer[Action, AuthRequest] {
```
Comment (ZsoltBalvanyos) [inline app/controllers/MockAuthRequest.scala]: This object is not used anymore, could be removed

File: build.sbt [hunk 3]
```diff
@@ -28,19 +28,22 @@ libraryDependencies ++= Seq(
...
+  "org.seleniumhq.selenium" % "selenium-chrome-driver" % "2.35.0" % Test exclude("org.slf4j", "log4j-over-slf4j"),
+  "org.json" % "json" % "20171018" % Test,
   "com.machinepublishers" % "jbrowserdriver" % "0.17.8" % Test exclude("org.slf4j", "log4j-over-slf4j"),
 
   // Integration test deps
-  "com.whisk" %% "docker-testkit-scalatest" % "0.9.3" % ServiceTest,
-  "com.whisk" %% "docker-testkit-impl-docker-java" % "0.9.3" % ServiceTest,
+  "com.whisk" %% "docker-testkit-scalatest" % "0.9.5" % ServiceTest,
+  "com.whisk" %% "docker-testkit-impl-docker-java" % "0.9.5" % ServiceTest,
+  "com.spotify" % "docker-client" % "8.9.2" % ServiceTest,
```
Comment (filosganga) [inline build.sbt]: Still using that?

File: conf/logback.xml [hunk 4]
```diff
@@ -12,6 +12,7 @@
     <logger name="org.apache" level="WARN" />
     <logger name="requestLogger" level="INFO"/>
     <logger name="com.google.api.client.http" level="INFO" />
+    <logger name="com.spotify.docker.client.LoggingPullHandler" level="OFF" />
```
Comment (filosganga) [inline conf/logback.xml]: Not needed if we don't use spotifyplugin

File: project/plugins.sbt [hunk 5]
```diff
@@ -1,6 +1,9 @@
...
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.6.7")
 addSbtPlugin("com.geirsson" % "sbt-scalafmt" % "0.5.6")
 addSbtPlugin("com.localytics" % "sbt-dynamodb" % "1.5.3")
 
 resolvers += Resolver.bintrayIvyRepo("ovotech", "sbt-plugins")
-addSbtPlugin("com.ovoenergy" % "sbt-comms-packaging" % "0.0.9")
+addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.3.2")
+addSbtPlugin("com.ovoenergy" % "sbt-comms-packaging" % "0.0.13")
+
+libraryDependencies += "com.spotify" % "docker-client" % "8.9.2"
```
Comment (filosganga) [inline project/plugins.sbt]: No needed anymore as we don't use it to publish

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 334](https://circleci.com/gh/ovotech/comms-audit-log/334) at 2017-11-23 11:42 GMT.

**Pull request merged**